# GWAS_script

### 1 Visualization_sweep_region.pl
visualization Fst and pi in cotton chrom.
Usage: perl $0 [options]

Options:
        
        -Fst        : Fst generated by vcftools      
        -Pi_group1   : Pi generated by vcftools       
        -Pi_group2   : Pi generated by vcftools

output: Creat Table and Figure of the top 1%, 5% sweep_region in cotton genome as output file automatically

require R Package ggplot2 and qqman
The version of cotton: Huyan et.al Nature genetic 2018 


### 2 Call snp on RNA seq data.pl

Usage: perl $0 [options]

Options:
        
        -sample        <s> : list of sample name, ${sample_name}.R1.clean.fastq.gz
        -ref            <s> : reference genome for mapping used hista2 index file, fasta [pwd]
        -gtf            <s> : gene transcription file [pwd]
        -thread         <i> : numters of thread to use [4]
        -bam          <i>: 0 , bam keeped; 1 ,bam will be deleted
require picard(V1.124),gatk(v-4.1.0.0),STAR(2.7.0d_0221)，samtools（V1.6）
output: Creat VCF file generated by GATK 4.0 as output file automatically


### 3 Stat_missing_ratio_on_vcf.r
Usage: Rscript ./out.imiss
The out.imiss generated by vcftools (vcftools --gzvcf input.vcf )
Options: None
output: Fig of missing ratio of 3 population cotton as output file automatically

### 4 Plotting Density Of Snps On Chromosomes.r
Usage: Rscript ./input.vcf

Options: None

output: Fig of density Of Snps On cotton Chromosomes as output file automatically and a table


### 5 Change_phylip_tree_name.pl
Usage: perl Change_phylip_tree_name.pl name.txt phylip.tree


